PhpAmqpLib/Wire/IO/StreamIO.php

```diff
@@ -224,10 +226,10 @@ public function read($len)
                 if ($this->canDispatchPcntlSignal) {
                     // prevent cpu from being consumed while waiting
                     if ($this->canSelectNull) {
-                        $this->select(null, null);
+                        $this->select(0, self::READ_BUFFER_WAIT_INTERVAL);
```

Comment:
Maybe instead of checking every `READ_BUFFER_WAIT_INTERVAL`µs, if the aim is to check for heartbeats absence, you can replace this line by something like 
```
$this->select($this->heartbeat === 0 ? null : $this->heartbeat, null);
```